### PR TITLE
Move AuditLogger interface to ecs-agent module

### PR DESCRIPTION
**Pull request by amogh09**
### Summary
Moving `AuditLogger` interface to the new `ecs-agent` module. The interface will be used in future functionality to be added to `ecs-agent` module.

### Implementation details
* Move `AuditLogger` interface and the corresponding `LogRequest` struct to `ecs-agent` module. 
* Update the mockgen configuration in `agent` module to not create mock implementation for `AuditLogger` interface.
* Add mockgen configuration to `ecs-agent` module to create mock implementation for `AuditLogger` interface in `ecs-agent` module. 
* Downgrade the version of testify used by `ecs-agent` to match the version used by `agent` module. 
* Make `agent` depend on `ecs-agent` and update the usage of `AuditLogger` and `LogRequest` to import them from `ecs-agent` module instead.

### Testing

Built Agent from source and performed the following tests that both trigger audit logging. 
1. Sent requests to credentials endpoint and checked that logs are emitted to `/var/log/ecs/audit.log` file. 
2. Exceeded request rate limit of TMDS and checked that requests rejected due to rate limits are logged to `/var/log/ecs/audit.log` file.

New tests cover the changes:  no

### Description for the changelog

Move AuditLogger interface to ecs-agent module.

### Licensing

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: ecs-agent/logger/audit/audit_log.go [hunk 1]
```diff
@@ -0,0 +1,24 @@
+// copyright amazon.com inc. or its affiliates. all rights reserved.
```
Comment (ohsoo) [inline ecs-agent/logger/audit/audit_log.go:1]: Just something I noticed, how come this copyright blurb doesn't have capitalization?

**Review by ohsoo — APPROVED** [review]

**Review by ohsoo — DISMISSED** [review]

File: agent/go.mod [hunk 2]
```diff
@@ -3,6 +3,7 @@ module github.com/aws/amazon-ecs-agent/agent
 go 1.19
 
 require (
+	github.com/aws/amazon-ecs-agent/ecs-agent v0.0.0-00010101000000-000000000000
```
Comment (fierlion) [inline agent/go.mod:6]: nit: was this go.mod auto-generated via `go mod` (I assume it was)?
can this just be `v0.0.0` since we're replacing it anyway?

**Review by fierlion — DISMISSED** [review]
one nit -- but not blocking on it.  If you can take a look in a future update, please do

File: agent/go.mod (hunk 2 above)
Comment (amogh09) [reply agent/go.mod]: Yeah this was auto-generated. I will simplify this.

**Review by fierlion — APPROVED** [review]
perfect in every way

**Review by mythri-garaga — APPROVED** [review]

**Pull request merged**